src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemQueryTagService.cs

```diff
@@ -5,23 +5,33 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using EnsureThat;
+using FellowOakDicom;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
 using Microsoft.Health.Dicom.Features.Common;
 
 namespace Microsoft.Health.Dicom.Core.Features.Workitem
 {
     public class WorkitemQueryTagService : IWorkitemQueryTagService, IDisposable
     {
         private readonly IIndexWorkitemStore _indexWorkitemStore;
-        private readonly AsyncCache<IReadOnlyCollection<WorkitemQueryTagStoreEntry>> _queryTagCache;
+        private readonly AsyncCache<IReadOnlyCollection<QueryTag>> _queryTagCache;
+        private readonly IDicomTagParser _dicomTagParser;
+        private readonly ILogger _logger;
         private bool _disposed;
 
-        public WorkitemQueryTagService(IIndexWorkitemStore indexWorkitemStore)
+        public WorkitemQueryTagService(IIndexWorkitemStore indexWorkitemStore, IDicomTagParser dicomTagParser, ILogger<WorkitemQueryTagService> logger)
         {
             _indexWorkitemStore = EnsureArg.IsNotNull(indexWorkitemStore, nameof(indexWorkitemStore));
-            _queryTagCache = new AsyncCache<IReadOnlyCollection<WorkitemQueryTagStoreEntry>>(ResolveQueryTagsAsync);
+            _queryTagCache = new AsyncCache<IReadOnlyCollection<QueryTag>>(ResolveQueryTagsAsync);
+            _dicomTagParser = EnsureArg.IsNotNull(dicomTagParser, nameof(dicomTagParser));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
         }
 
         public void Dispose()
```

Comment:
nit: in general, if this class is `sealed` because we don't foresee derived implementations, we don't need to implement the entire disposal pattern and instead can have a simply `Dispose` method, like:
```csharp
public void Dispose()
{
    _queryTagCache.Dispose();
    GC.SuppressFinalize(this);
}
```

Reply:
`AsyncCache<T>` will also throw an `ObjectDisposedException`, so you don't need to maintain your own `_disposed` flag

Reply:
Done